ngrx/ducks-schematics/src/collection/duck/files/__name@dasherize@if-flat__/__name@dasherize__.effects.ts.template

```diff
@@ -0,0 +1,20 @@
+import { Inject, Injectable } from '@angular/core';
+import { Duck, whereType } from '@co-it/ngrx-ducks';
+import { Actions, Effect } from '@ngrx/effects';
+import { delay, map } from 'rxjs/operators';
+import { <%= classify(name) %> } from './<%= dasherize(name) %>.duck';
+
+@Injectable()
+export class <%= classify(name) %>Effects {
+  @Effect()
```

Comment:
Can you add a comment that this is an example?
I often see that generated code is not replaced by developers and remains unless it is already released.

```ts
/**
 * Generated example
 *
 * You may want to replace this Effect with your own implementation.
 */

```

Reply:
I added the comment, and I will try to add the Effect to the nearest module and/or add an extra parameter for it - a behavior like `ng generate module` would be great I think ...

Reply:
kinda tricky to check and add it to an existing `EffectsModule.forFeature([SomeOtherEffects])` entry in the nearest, or via params passed, module - else import `EffectsModule` from ngrx and add the whole entry `EffectsModule.forFeature([MyDuckEffects])`, also have to check for `EffectsModule.forRoot(...)` I guess (add as root if no root was found?) ... is there a public API to help out here? 🤔 I am copying + adopting the angular schematics sources, feels a bit strange ... but I think I can handle it ... maybe it is a good idea to also check the ngrx schematics sources ... definitely a good place to learn a lot I guess ...

... and how to handle the selectors? leave it like it is, or register the reducer too, with some options? maybe splitting the schematic is a better option?

Reply:
## Effect

It should be the same behaviour as adding a generated Component to a module file.

I did not implement this before but the API you may have to look for is called `recorder` (see: https://www.softwarearchitekt.at/post/2018/03/20/custom-schematics-part-iv-frictionless-library-setup-with-the-angular-cli-and-schematics.aspx).

We can also postpone this feature to a later release. :-)

What do you think?

## Selectors

For now, I would leave it as is.
In general, I prefer to have separate schematics for the effect and the selector.

Reply:
thx for the link :+1: I will check it out and come back afterwards (soon hopefully)